ppx/test/example.ml

```diff
@@ -21,6 +21,14 @@ type allow_extra_fields = {a: int} [@@deriving json] [@@json.allow_extra_fields]
 type allow_extra_fields2 = A of {a: int} [@json.allow_extra_fields] [@@deriving json]
 type drop_default_option = { a: int; b_opt: int option; [@option] [@json.drop_default] } [@@deriving json]
 
+(*module Polyvar : sig*)
+(*  type t = [`A | `B] [@@deriving json]*)
+(*end = struct*)
+(*  type t = [`A | `B] [@@deriving json]*)
+(*end*)
+(**)
+(*type polyvar = [Polyvar.t|`C] [@@deriving json]*)
```

Comment:
Should we add some test for the poly compose case? e.g.

```
type polycompose = [poly2 | `C] [@@deriving json]
```

then:
```
  C ({|["P2", 42, "hello"]|}, polycompose_of_json, polycompose_to_json, (`P2 (42, "hello") : polycompose));
  C ({|["C"]|}, polycompose_of_json, polycompose_to_json, (`C : polycompose));
```

Reply:
oh right. the `type poly = [ `A | `B of int | other ] [@@deriving json]`. i missed it.